### PR TITLE
Add zone to q_options in call_automate.

### PR DESCRIPTION
**Pull request by tinaafitz**
Separate Manageiq change to pass zone_name required.

https://bugzilla.redhat.com/show_bug.cgi?id=1447625

Related PR https://github.com/ManageIQ/manageiq/pull/15026

### CODE REVIEW

File: spec/miq_ae_event_spec.rb [hunk 1]
```diff
@@ -65,10 +65,31 @@
 
       context "with user owned VM" do
         let(:vm_owner) { user }
+        let(:miq_server) { EvmSpecHelper.local_miq_server(:is_master => true) }
+        let(:zone) { FactoryGirl.create(:zone) }
 
-        it "has tenant" do
+        it "raise_evm_event, deliver_queue, with zone" do
```
Comment (mkanoor) [inline spec/miq_ae_event_spec.rb]: @tinaafitz can we use a shared example for these 2 tests and just change the zones

File: spec/miq_ae_event_spec.rb [hunk 2]
```diff
@@ -80,7 +101,7 @@
         it "has tenant" do
           args = {:user_id => admin.id, :miq_group_id => vm_group.id, :tenant_id => vm_group.tenant.id}
           expect(MiqAeEngine).to receive(:deliver_queue).with(hash_including(args), anything)
-
+          expect(MiqServer).to receive(:my_zone).and_return('default')
```
Comment (mkanoor) [inline spec/miq_ae_event_spec.rb]: @tinaafitz 
Can we have something like this
before { allow(MiqServer).to receive_messages(:my_zone => "default") }
then we dont have to change every line

Comment (tinaafitz) [inline spec/miq_ae_event_spec.rb]: @mkanoor As discussed, will refactor this test in a separate PR.

File: lib/miq_automation_engine/engine/miq_ae_event.rb [hunk 3]
```diff
@@ -121,6 +121,7 @@ def self.call_automate(obj, attrs, instance_name, options = {})
     raise "A user is needed to raise #{instance_name} to automate. [#{obj.class.name}] id:[#{obj.id}]" unless user
 
     q_options = {
+      :zone         => Zone.determine_queue_zone(options),
```
Comment (gmcculloug) [inline lib/miq_automation_engine/engine/miq_ae_event.rb]: @tinaafitz This is already called during `MiqQueue.put`?  Is this really needed?

File: lib/miq_automation_engine/engine/miq_ae_event.rb [hunk 4]
```diff
@@ -125,6 +125,7 @@ def self.call_automate(obj, attrs, instance_name, options = {})
       :priority     => MiqQueue::HIGH_PRIORITY,
       :task_id      => nil          # Clear task_id to allow running synchronously under current worker process
     }
+    q_options[:zone] = options[:zone] if options.keys.include?(:zone) && !options[:zone].nil?
```
Comment (gmcculloug) [inline lib/miq_automation_engine/engine/miq_ae_event.rb]: `q_options[:zone] = options[:zone] if options[:zone].present?`

File: spec/miq_ae_event_spec.rb [hunk 5]
```diff
@@ -63,24 +62,65 @@
...
+        it_behaves_like "#zone not passed to raise_evm_event"
+      end
+
       context "with group owned VM" do
         let(:vm_group) { FactoryGirl.create(:miq_group, :tenant => FactoryGirl.create(:tenant)) }
 
         it "has tenant" do
           args = {:user_id => admin.id, :miq_group_id => vm_group.id, :tenant_id => vm_group.tenant.id}
           expect(MiqAeEngine).to receive(:deliver_queue).with(hash_including(args), anything)
-
```
Comment (gmcculloug) [inline spec/miq_ae_event_spec.rb]: @tinaafitz No sure if you intentionally remove this line but it seems random with the other changes in this PR and now makes this test inconsistent with (most) of the remaining tests that separate the `expect().to receive` and the following method call.  I tend to prefer this separation between testing logic and the code being run for the test.

Comment (tinaafitz) [inline spec/miq_ae_event_spec.rb]: @gmcculloug Added blank line back, thanks for catching it.

**Comment by simaishi** [comment]
Fine backport details:

```
$ git log -1
commit 86c23029db18065a2ea5b27d00aac60fc170c518
Author: Greg McCullough <gmccullo@redhat.com>
Date:   Wed May 10 10:23:00 2017 -0400

    Merge pull request #18 from tinaafitz/add_zone_to_qoptions
    
    Add zone to q_options in call_automate.
    (cherry picked from commit 0495c11579d5751f91c6e6791c65cdd1db09ad88)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1449748
```

**Comment by simaishi** [comment]
Euwe backport (to manageiq repo) details:

```
$ git log -1
commit 7051ce276be724232cd2d0b4e46b31e8e24f2ab1
Author: Greg McCullough <gmccullo@redhat.com>
Date:   Wed May 10 10:23:00 2017 -0400

    Merge pull request #18 from tinaafitz/add_zone_to_qoptions
    
    Add zone to q_options in call_automate.
    (cherry picked from commit 0495c11579d5751f91c6e6791c65cdd1db09ad88)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1449753
```

**Pull request merged**